src/network/networkd-link.c

```diff
@@ -99,6 +99,21 @@ static bool link_ipv6ll_enabled(Link *link) {
         return link->network->link_local & ADDRESS_FAMILY_IPV6;
 }
 
+static bool link_ipv6_enabled(Link *link) {
+        assert(link);
+
+        if (!socket_ipv6_is_supported())
+                return false;
+
+        if (link_dhcp6_enabled(link) && link_ipv6ll_enabled(link))
+                return true;
```

Comment:
Well @poettering  suggested it If I got that right 

"return false if socket_ipv6_is_supported() returns false, or when both of link_dhcp6_enabled() + link_ipv6ll_enabled() return false and if there's no static ipv6 address configured either."
https://github.com/systemd/systemd/pull/3058